### PR TITLE
Resolving #279 - improve build summary

### PR DESCRIPTION
**Pull request by huanlin**

### CODE REVIEW

**Comment by mzorec** [comment]
Thank you for PR!

-  I think that target's that were not run because previous target failed should also be displayed in build summary.

```c#
	var fail = session.CreateTarget("test").Do(Test);

            context.CreateTarget("Rebuild")
                .SetDescription("Rebuilds the solution and runs tests.")
                .SetAsDefault()
                .DependsOn(compile, fail, tests);
```
I have executed rebuild and got following output (fail target failed):

image

I think that "tests" target should also be displayed with Not ran status.

- Targets that were skipped because when condition was not satisfied should also be displayed in build summary with status skipped.

```C#
     var skiped = context.CreateTarget("Skipped").Do((c) => { context.LogInfo("..."); });

      context.CreateTarget("Sample")
           .DependsOn(TargetA)
           .DependsOn(skiped).When((c) => 1 == 2)
           .DependsOn(TargetB);
```

Skiped target will not be executed when you execute target Sample because when condition is never meet but target should be displayed in build summary.

**Comment by huanlin** [comment]
> Skiped target will not be executed when you execute target Sample because when condition is never meet but target should be displayed in build summary.

Thanks for detailed descriptions and suggestions. I'll try to fix it.

File: src/FlubuCore/Targeting/TargetTree.cs [hunk 1]
```diff
@@ -131,7 +139,6 @@ public virtual void EnsureDependenciesExecuted(ITaskContextInternal taskContext,
         public virtual void ResetTargetTree()
         {
             _targets.Clear();
-            _executedTargets.Clear();
```
Comment (mzorec) [inline src/FlubuCore/Targeting/TargetTree.cs:134]: Why was this line removed?

Comment (huanlin) [reply src/FlubuCore/Targeting/TargetTree.cs:134]: Because .the next line is calling `ResetTargetExecutionInfo` method and that method has already do the same thing.

```
        public virtual void ResetTargetTree()
        {
            _targets.Clear();
            ResetTargetExecutionInfo();
            AddDefaultTargets();
        }

        public void ResetTargetExecutionInfo()
        {
            _executedTargets.Clear();
        }
```

**Comment by mzorec** [comment]
Everything looks ok now. Thank you for PR!
